code/game/objects/structures/grille.dm

```diff
@@ -49,17 +49,19 @@
 		new /obj/structure/grille/ratvar(src.loc)
 	qdel(src)
 
-/obj/structure/grille/Bumped(atom/user)
-	if(ismob(user))
-		var/tile_density = FALSE
-		for(var/atom/movable/AM in get_turf(src))
-			if(AM == src)
-				continue
-			if(AM.density && AM.layer >= layer)
-				tile_density = TRUE
-				break
-		if(!tile_density)
-			shock(user, 70)
+/obj/structure/grille/CollidedWith(atom/movable/AM)
+	if(!ismob(AM))
+		return
+	var/mob/M = AM
+	var/tile_density = FALSE
+	for(var/atom/movable/TAM in get_turf(src))
```

Comment:
So this is 1: double bumping, and two, also calling on bumped(), so thats 3 times grills shock you?

Am I understanding that right?

wow,